src/core/numerics/pusher/boris.hpp

```diff
@@ -151,7 +176,7 @@ class BorisPusher
      * @return the function returns and iterator on the first leaving particle, as
      * detected by the ParticleSelector
      */
-    void prePushStep_(ParticleRange const& rangeIn, ParticleRange& rangeOut)
+    void prePushStep_(ParticleRange const& rangeIn, ParticleRange& rangeOut, int alpha = 1)
```

Comment:
I understand `alpha` is a coefficient to get `dt` from `dt/2`.
This works, but reads weird. One cannot guess what alpha is without seing where `alpha` being used and here `alpha` only appears as a "workaround" to "fix" an implementation that internally has dt/2.

I would probably think about something that would change the half time step to a regular time step and use a factor of 0.5 only for half pushes.